R/print.data.table.R

```diff
@@ -101,7 +101,22 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
       IDate = "<IDat>", integer64 = "<i64>", raw = "<raw>",
       expression = "<expr>", ordered = "<ord>")
     classes = classes1(x)
-    abbs = unname(class_abb[classes])
+    col_names <- colnames(toprint)
+    index_cols <- paste0("index:", indices(x))
+    for (col_name in col_names) {
+      if (col_name %in% index_cols) {
+        classes[col_name] <- "index"
+      } else if (col_name %in% names(x)) {
+        cls <- class(x[[col_name]])
+        if (is.list(cls)) cls <- unlist(cls)
+        if (length(cls) == 0) cls <- "unknown"
```

Comment:
I don't think R will allow making the class vector a list. Also, `class()` will return the name of the primitive type of the value if the class attribute is set empty or removed altogether, so both tests here are impossible.